fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/models/ResourceFactoryImpl.java

```diff
@@ -150,6 +150,42 @@ public FedoraResource getContainer(final String transactionId, final FedoraId re
         }
     }
 
+    @Override
+    public String getInteractionModel(final String transactionId, final FedoraId fedoraId) {
+        if (fedoraId.isRepositoryRoot()) {
+            return BASIC_CONTAINER.getURI();
+        }
+        if (!(fedoraId.isMemento() || fedoraId.isAcl())) {
+            // containment index doesn't handle versions and only tells us if the resource (not acl) is there,
+            // so don't bother checking for them.
+            return containmentIndex.getInteractionModel(transactionId, fedoraId);
```

Comment:
I'm not sure if I follow why the interaction model is being added to the containment index. Is it just a place to stick it so that the resource headers don't need to be read?

Reply:
Hm... I think we should think about this more. It feels like data is being tacked on here just because it's a place to put it.

In the cases where `getInteractionModel()` is used, how often are the ResourceHeaders still loaded as part of processing the request? If the headers are still being loaded most of the time, then it would seem to me that a better optimization would be to cache the resource headers rather than sticking the interaction model in the database.

If we do want this information in the database, then I think we should consider instead putting all of the resource headers into the database as a new table.

Reply:
Ok containment index is left alone.